### ISSUE TITLE
Error "No OAuth token with Google Drive scope was found" When Connecting to BigQuery

### ISSUE THREAD

**Issue by tamvm** [metabase/metabase#13001]
**Describe the bug**
Can connect from metabase to Bigquery but can't allow to load table which datasource is from Google Spreadsheet. 

**Logs**
https://gist.githubusercontent.com/tamvm/cfaacc30b30dff3d12dbdaa930eb04eb/raw/256892e09d10addbc8e38c78eb4ff53d2fbd06b2/metabase-bigquery-spreadsheet.log

**To Reproduce**
1. Follow this instruction to craete table from Google Spreadsheet in BigQuery
https://torbjornzetterlund.com/creating-a-table-in-bigquery-with-your-google-sheet-data/

2. Create Service Account in Google Cloud and create database connection in Metabase

3. Query table in Step 1. then the error "No OAuth token with Google Drive scope was found" will be shown (a attached image below)

**Expected behavior**
By default, 

**Screenshots**
![image](https://user-images.githubusercontent.com/344684/88511328-f15e1c00-d027-11ea-8f77-4cecc84c30fa.png)

Information about your Metabase Installation:

You can get this information by going to Admin -> Troubleshooting.

Your browser and the version: Version 84.0.4147.89 (Official Build) (64-bit)
Your operating system: MacOS
Your databases: BigQuery
Metabase version: 3.6
Metabase hosting environment: Docker
Metabase internal database: Postgres
Severity
How severe an issue is this bug to you? Is this annoying, blocking some users, blocking an upgrade or blocking your usage of Metabase entirely?
Medium priority

**Another context**
I think it's related to scopes used. I think it should allow user to authorize Google Drive scope: https://developers.google.com/drive/api/v2/reference/permissions/get to be able to use Google Drive API 

---
:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by flamber** [comment]
Just for reference, the old method had a separate link for giving Google Drive permissions:
![image](https://user-images.githubusercontent.com/1447303/88812257-d0c4cc00-d1b7-11ea-9995-c283c1197847.png)

Not sure if we just need to document this better or have an extra option somewhere. It seems like there's a workaround:
https://stackoverflow.com/questions/40391128/how-do-i-authenticate-a-service-account-to-make-queries-against-a-gdrive-sheet-b

**Comment by jmcdole** [comment]
A view and table trigger different error messages.  For me, the view was an OAuth error.  And the table was a permissions error.

**Comment by tamvm** [comment]
@ghost yes, it may raise "Failed to read the spreadsheet. Error code: PERMISSION_DENIED", I think it's same issue with this one

**Comment by marcoruggine** [comment]
I'm stuck on the same PERMISSION_DENIED issue

**Comment by gabrielvenancioo** [comment]
I'm also stuck on "Failed to read the spreadsheet. Error code: PERMISSION_DENIED"

**Comment by marcoruggine** [comment]
Hi guys, do you have any plans on working on this? I could manually load a bunch of CSVs in BQ but I'd rather not do any manual labor since it would mean losing synchronicity

**Comment by tamvm** [comment]
For now, you can work around by using Query Settings in BQ and save in new table, then connect that table to Metabase.

**Comment by marcbllv** [comment]
Any plans on working on this ? That would be great to add the possibility to set specific scopes for BigQuery connections!

**Comment by asyarif93** [comment]
any news of this case?

in java scopes can be added this way
```
GoogleCredentials credentials =
    ServiceAccountCredentials.getApplicationDefault()
        .createScoped(
            ImmutableSet.of(
                "https://www.googleapis.com/auth/bigquery",
                "https://www.googleapis.com/auth/drive"));
```

I'm unfamiliar with Clojure so I can't help make any contribution. and I'm not sure to make scope that way, as in general some of Service Account might not have drive permission.

**Comment by AyrtonB** [comment]
How can we add scopes through Metabase?

When I connect the BigQuery table to metabase it says `Success!` and I can see through the data model that its been bale to pick up types, names, etc. However when I go to browse data I get a similar error:

`403 Forbidden POST https://www.googleapis.com/bigquery/v2/projects/project/queries { "code" : 403, "errors" : [ { "domain" : "global", "message" : "Access Denied: BigQuery BigQuery: Permission denied while getting Drive credentials.", "reason" : "accessDenied" } ], "message" : "Access Denied: BigQuery BigQuery: Permission denied while getting Drive credentials.", "status" : "PERMISSION_DENIED" }`

**Comment by flamber** [comment]
Closing, as 0.42 will include Google Drive scope in the new BigQuery driver. And the old OAuth driver is deprecated.
For reference #4179 and #19629

**Comment by n-kostadinov** [comment]
> How can we add scopes through Metabase?
> 
> When I connect the BigQuery table to metabase it says `Success!` and I can see through the data model that its been able to pick up types, names, etc. However, when I go to browse data I get a similar error:
> 
> `403 Forbidden POST https://www.googleapis.com/bigquery/v2/projects/project/queries { "code" : 403, "errors" : [ { "domain" : "global", "message" : "Access Denied: BigQuery BigQuery: Permission denied while getting Drive credentials.", "reason" : "accessDenied" } ], "message" : "Access Denied: BigQuery BigQuery: Permission denied while getting Drive credentials.", "status" : "PERMISSION_DENIED" }`
> 
> I've added the relevant roles to the service account, enabled the service account to edit the google sheet big query is running from, and enabled the google drive API. I've also enabled Google Workspace Domain-wide Delegation with the scopes mentioned for the service worker that metabase is using.

I did all those things too. Service Account is Big Query Admin. Has domain-wide delegation with the two scopes mentioned. The sheet is shared with the service account and also with very one in the domain. We have Google Business Suite. I am too getting: 

403 Forbidden POST https://www.googleapis.com/bigquery/v2/projects/perty-metabase/queries { "code" : 403, "errors" : [ { "domain" : "global", "message" : "Access Denied: BigQuery BigQuery: Permission denied while getting Drive credentials.", "reason" : "accessDenied" } ], "message" : "Access Denied: BigQuery BigQuery: Permission denied while getting Drive credentials.", "status" : "PERMISSION_DENIED"

**Comment by flamber** [comment]
@n-kostadinov Unless you are testing 0.42.0-rc2, then that would be expected. Otherwise open a new issue.

**Comment by n-kostadinov** [comment]
> 

Hi, I am using 0.41.5 I will wait for the official Release of 0.42 Many thanks for your work.

**Comment by n-kostadinov** [comment]
FYI, issue has been resolved in 0.42.0. 
At first I got the same error as before. The thing that I was missing was actually sharing the Google Drive Sheet with the service account. Hope this helps others.
